apps/docs/pages/docs/api-reference/fields/custom.mdx

```diff
@@ -211,6 +212,156 @@ const config = {
   <Puck.Preview />
 </ConfigPreview>
 
+### key
+
+Assign a unique key to force the field to remount when resolving different custom field types on the same prop.
+
+```tsx {17, 19, 35, 37} showLineNumbers copy
+const config = {
+  components: {
+    Example: {
+      fields: {
+        isText: {
+          type: "radio",
+          options: [
+            { label: "Yes", value: true },
+            { label: "No", value: false },
+          ],
+        },
+      },
+      resolveFields: (data, params) => {
+        if (data.props.isText) {
+          return {
+            ...params.fields,
+            title: {
```

Comment:
Suggest you use a different name for the prop, as a number "title" doesn't make sense. Perhaps `prop` is fine.